### PR TITLE
Adds Room to todo-mvp-clean and updates versions

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: todoapp/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -15,9 +14,9 @@
   ~ limitations under the License.
   -->
 
-<manifest
-    package="com.example.android.architecture.blueprints.todoapp"
-    xmlns:android="http://schemas.android.com/apk/res/android">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (florina-muntenescu) [inline todoapp/app/src/main/AndroidManifest.xml]: why is tools needed?

File: todoapp/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -26,16 +25,24 @@
...
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
         <activity android:name="com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailActivity" />
         <activity android:name="com.example.android.architecture.blueprints.todoapp.addedittask.AddEditTaskActivity" />
-        <activity android:name="com.example.android.architecture.blueprints.todoapp.statistics.StatisticsActivity" />
+        <activity
+            android:name="com.example.android.architecture.blueprints.todoapp.statistics.StatisticsActivity"
+            android:parentActivityName=".tasks.TasksActivity"
+            tools:ignore="UnusedAttribute">
```
Comment (florina-muntenescu) [inline todoapp/app/src/main/AndroidManifest.xml]: is this also needed?

File: .travis.yml [hunk 3]
```diff
@@ -3,9 +3,10 @@ android:
   components:
     - tools
     - platform-tools
-    - build-tools-24.0.2
-    - android-24
+    - build-tools-26.0.2
+    - android-26
```
Comment (piotrek1543) [inline .travis.yml]: Is there any reason why don't you use the newest libraries version, I mean: android-27, build-tools-27.0.0 and Gradle 4.3?

File: todoapp/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (piotrek1543) [reply todoapp/app/src/main/AndroidManifest.xml]: @florina-muntenescu tools in xml files are used to fix any Android Lint issue like "app is not indexed by Google" or no backup data preference. Here, this attribute is used only to fix "UnusedAttribute" Android Lint issue. 
Disclaimer: I suppose you 

I would wonder rather why this issue was fixed, than `why is tools needed` :-)

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java [hunk 4]
```diff
@@ -43,39 +50,60 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        mActionBar.setDisplayHomeAsUpEnabled(true);
+        mActionBar.setDisplayShowHomeEnabled(true);
 
-        AddEditTaskFragment addEditTaskFragment =
-                (AddEditTaskFragment) getSupportFragmentManager().findFragmentById(
-                        R.id.contentFrame);
+        AddEditTaskFragment addEditTaskFragment = (AddEditTaskFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.contentFrame);
 
         String taskId = getIntent().getStringExtra(AddEditTaskFragment.ARGUMENT_EDIT_TASK_ID);
```
Comment (piotrek1543) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java]: What about to change an order of lines `57` and `60`. Lines above first mentioned are about customizing `Toolbar` widget and lines below the second are about providing fragments to `Activity`. Now it looks a bit like toolbar->fragment->toolbar->fragment

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/ToDoDatabase.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import com.example.android.architecture.blueprints.todoapp.tasks.domain.model.Task;
+
+
+/**
+ * The Room Database that contains the Task table.
+ */
+@Database(entities = {Task.class}, version = 1)
+public abstract class ToDoDatabase extends RoomDatabase {
+
+    private static ToDoDatabase INSTANCE;
```
Comment (piotrek1543) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/ToDoDatabase.java]: I wonder why this variable name is uppercase when it is not a final, just static one?

File: todoapp/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/AndroidManifest.xml]: Yes

File: todoapp/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/AndroidManifest.xml]: What Piotrek said.

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/ToDoDatabase.java (hunk 5 above)
Comment (JoseAlcerreca) [reply todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/ToDoDatabase.java]: You are right but we need to do that in todo-mvp

**Pull request merged**